Failure to load markdown viewer on MS-Windows

Hi, there is an issue trying to load the markdown viewer on <S-Windows.

To reproduce, just run the markdown test suite

```batch
>powershell -command clojure -X:test/markdown
....
Execution error (PolyglotException) at sun.nio.fs.WindowsPathParser/normalize (WindowsPathParser.java:182).
java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/src/viewers/modules/markdown/resources/js/markdown.mjs
```
This is due to an invalid path generated to `markdown.mjs` passed in the graalvm polygot.

Also, the `clojure` cli tool can't seem to be able to download the `clerk` dependency from the `devdocs` module.

```batch
>powershell -command clojure -Stree
Cloning: git@github.com:nextjournal/clerk.git
Error building classpath. Unable to clone C:\Users\ikappaki\.gitlibs\_repos\git@github.com\nextjournal\clerk
FATAL ERROR: No supported authentication methods available (server sent: publickey)
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

This appears to be due to the `git:/` protocol used for that particular dependency.

PR to follow, thanks.

As reported on Slack nextjournal this also causes a startup problem with clerk-demo on Windows:
I've got a small Clerk project I created already running ok on my system.
I'm trying to start:
https://github.com/nextjournal/clerk-demo
from a Windows terminal using "clj" (CLI version 1.10.3.933).
I get a startup exception:
Caused by: org.graalvm.polyglot.PolyglotException: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/Users/mail/.gitlibs/libs/io.github.nextjournal/viewers/1b1374db20b56d14a0f32835c1ce5d9254c23527/modules/markdown/resources/js/markdown.mjs

--- a/modules/devdocs/deps.edn
+++ b/modules/devdocs/deps.edn
@@ -6,8 +6,7 @@
         io.github.nextjournal/cljs-extensions {:local/root "../cljs-extensions"}
         io.github.nextjournal/markdown {:local/root "../markdown"}
 
-        com.nextjournal/clerk {:git/url "git@github.com:nextjournal/clerk.git"
-                               :git/sha "d2cacf6d3d27768f34b9092c100be9083c6dd1fc"}
+        io.github.nextjournal/clerk {:git/sha "ebc8f998c761f1b0e2c9fc4750ae87d8f1766faa"}
 
         ;; we use shadow-resource directly
         thheller/shadow-cljs {:mvn/version "2.15.12"}

--- a/modules/markdown/build.clj
+++ b/modules/markdown/build.clj
@@ -1,0 +1,31 @@
+(ns build
+  (:require [clojure.tools.build.api :as b]
+            [deps-deploy.deps-deploy :as dd]))
+
+(def lib 'io.github.nextjournal/markdown)
+(def version (format "0.1.%s" (b/git-count-revs nil)))
+(def class-dir "target/classes")
+(def basis (b/create-basis {:project "deps.edn"}))
+(def jar-file (format "target/%s-%s.jar" (name lib) version))
+
+(defn clean [_]
+  (b/delete {:path "target"}))
+
+(defn jar [_]
+  (b/write-pom {:class-dir class-dir
+                :lib lib
+                :version version
+                :basis basis
+                :src-dirs ["src"]})
+  (b/copy-dir {:src-dirs ["src" "resources"]
+               :target-dir class-dir})
+  (b/jar {:class-dir class-dir
+          :jar-file jar-file}))
+
+(defn deploy [opts]
+  (println "Deploying version" jar-file "to Clojars.")
+  (jar {})
+  (dd/deploy (merge {:installer :remote
+                     :artifact jar-file
+                     :pom-file (b/pom-path {:lib lib :class-dir class-dir})}
+                    opts)))

--- a/modules/markdown/deps.edn
+++ b/modules/markdown/deps.edn
@@ -3,4 +3,8 @@
         lambdaisland/uri {:mvn/version "1.10.79"}
         org.clojure/data.json {:mvn/version "0.2.6"}
         org.graalvm.js/js {:mvn/version "21.3.0"}
-        org.graalvm.js/js-scriptengine {:mvn/version "21.3.0"}}}
+        org.graalvm.js/js-scriptengine {:mvn/version "21.3.0"}}
+
+ :aliases {:build {:deps {io.github.clojure/tools.build {:git/tag "v0.6.1" :git/sha "515b334"}
+                          io.github.slipset/deps-deploy {:git/sha "b4359c5d67ca002d9ed0c4b41b710d7e5a82e3bf"}}
+                   :ns-default build}}}

--- a/modules/markdown/deps.edn
+++ b/modules/markdown/deps.edn
@@ -2,5 +2,5 @@
  :deps {applied-science/js-interop {:mvn/version "0.2.6"}
         lambdaisland/uri {:mvn/version "1.10.79"}
         org.clojure/data.json {:mvn/version "0.2.6"}
-        org.graalvm.js/js {:mvn/version "21.2.0"}
-        org.graalvm.js/js-scriptengine {:mvn/version "21.2.0"}}}
+        org.graalvm.js/js {:mvn/version "21.3.0"}
+        org.graalvm.js/js-scriptengine {:mvn/version "21.3.0"}}}

--- a/modules/markdown/deps.edn
+++ b/modules/markdown/deps.edn
@@ -3,5 +3,4 @@
         lambdaisland/uri {:mvn/version "1.10.79"}
         org.clojure/data.json {:mvn/version "0.2.6"}
         org.graalvm.js/js {:mvn/version "21.2.0"}
-        org.graalvm.js/js-scriptengine {:mvn/version "21.2.0"}
-        nextjournal/log {:local/root "../log"}}}
+        org.graalvm.js/js-scriptengine {:mvn/version "21.2.0"}}}

--- a/modules/markdown/src/nextjournal/markdown.clj
+++ b/modules/markdown/src/nextjournal/markdown.clj
@@ -4,29 +4,29 @@
             [clojure.data.json :as json]
             [nextjournal.markdown.parser :as markdown.parser]
             [nextjournal.markdown.transform :as markdown.transform])
-  (:import [org.graalvm.polyglot Context Context$Builder Engine Source Value]))
+  (:import [org.graalvm.polyglot Context Context$Builder Source Value]))
 
 (set! *warn-on-reflection* true)
 
-(def ^Engine engine (Engine/create))
-
 (def ^Context$Builder context-builder
   (doto (Context/newBuilder (into-array String ["js"]))
-    (.engine engine)
     (.option "js.timer-resolution" "1")
     (.option "js.java-package-globals" "false")
+    (.option "js.esm-eval-returns-exports", "true") ;; returns module exports when evaling an esm module file
     (.out System/out)
     (.err System/err)
+    (.allowIO true)
     (.allowExperimentalOptions true)
     (.allowAllAccess true)
-    (.allowNativeAccess true)))
+    (.allowNativeAccess true)
+    (.option "engine.WarnInterpreterOnly" "false")))
 
 (def ^Context ctx (.build context-builder))
 
 (def ^Value MD-imports
-  (.eval ctx (.build (Source/newBuilder "js"
-                                        (str "import MD from '" (.getPath (io/resource "js/markdown.mjs")) "'; MD")
-                                        "source.mjs"))))
+  (.. ctx
+      (eval (.build (Source/newBuilder "js" (io/resource "js/markdown.mjs"))))
+      (getMember "default")))
 
 (defn make-js-fn [fn-name]
   (let [f (.getMember MD-imports fn-name)]

--- a/modules/markdown/src/nextjournal/markdown/parser.cljc
+++ b/modules/markdown/src/nextjournal/markdown/parser.cljc
@@ -14,8 +14,8 @@
   "
   (:require [clojure.string :as str]
             [nextjournal.markdown.transform :as md.transform]
-            [nextjournal.log :as log]
-            #?(:cljs [applied-science.js-interop :as j])))
+            #?@(:cljs [[applied-science.js-interop :as j]
+                       [cljs.reader :as reader]])))
 
 ;; clj common accessors
 (def get-in* #?(:clj get-in :cljs j/get-in))
@@ -31,7 +31,7 @@
 ;; region node operations
 ;; helpers
 (defn inc-last [path] (update path (dec (count path)) inc))
-(defn hlevel [{:as _token hn :tag}] (when (string? hn) (some-> (re-matches #"h([\d])" hn) second #?(:clj read-string :cljs cljs.reader/read-string))))
+(defn hlevel [{:as _token hn :tag}] (when (string? hn) (some-> (re-matches #"h([\d])" hn) second #?(:clj read-string :cljs reader/read-string))))
 (defn parse-fence-info
   "Ingests nextjournal, GFM, Pandoc and RMarkdown fenced code block info, returns a map
 
@@ -208,7 +208,7 @@ end"
 (declare apply-tokens)
 (defmulti apply-token (fn [_doc token] (:type token)))
 (defmethod apply-token :default [doc token]
-  (log/warn :apply-token/unknown-type {:token token})
+  (prn :apply-token/unknown-type {:token token})
   doc)
 
 ;; blocks

--- a/modules/view/deps.edn
+++ b/modules/view/deps.edn
@@ -1,2 +1,2 @@
 {:deps {nextjournal/cljs-extensions {:local/root "../cljs-extensions"}
-        nextjournal/log             {:local/root "../log"}}}
+        io.github.nextjournal/log   {:local/root "../log"}}}

